src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -78,6 +80,20 @@ def config_dashboard(self, daemon_descrs: List[DaemonDescription]) -> None:
             service_url
         )
 
+    def ok_to_stop(self, daemon_ids: List[str], force: Optional[bool] = False) -> HandleCommandResult:
```

Comment:
Using the proposed "is_safe_ok_to_stop"  CephadmService  method we can simplify the ok_to_stop methods for specific services, for example for Grafana (at least 1 daemon needed to provide service)

```python
        warn, warn_message = self.is_safe_ok_to_stop(self.TYPE, daemon_ids, 'Grafana', 1)
        if warn and not force:
            return HandleCommandResult(1, None, warn_message)
        return HandleCommandResult(0, warn_message, None)
```